src/main/java/org/folio/coursereserves/util/Util.java

```diff
@@ -8,4 +8,18 @@ public static String logAndSaveError(Throwable err, Logger logger) {
     logger.error(message, err);
     return message;
   }
+
+  /**
+   * Combine limit query and optional user query.
+   * @param userQuery any query provided by API/user; null for none
+   * @param limitQuery simple field=value query
+   * @return combined CQL query.
+   */
+  public static String getQueryWithLimit(String userQuery, String limitQuery) {
+    if (userQuery == null || userQuery.isEmpty()) {
+      return limitQuery;
+    } else {
+      return limitQuery + " AND (" + userQuery + ")";
+    }
+  }
```

Comment:
The APIs have a limit=10 parameter, this makes the limit in `limitQuery` somewhat misleading for me.
If both sub-queries are put into braces people can use OR in both of them.
How about a general-purpose method with each sub-query being optional?

```
  public static String cqlAnd(String subQuery1, String subQuery2) {
    if (StringUtils.isBlank(subQuery1)) {
      if (StringUtils.isBlank(subQuery2)) {
        return "cql.allRecords=1";
      } else {
        return subQuery2;
      }
    } else {
      if (StringUtils.isBlank(subQuery2)) {
        return subQuery1;
      } else {
        return "(" + subQuery1 + ") AND (" + subQuery2 + ")";
      }
    }
  }
```

Reply:
People can not put the limitQuery in braces or anything like that. It's a simple field=<value> where the value is a path component , like listening_id. Therefore the implementation I chose was the minimal to support that.